### PR TITLE
PySpark sort by columns and DataFrame.getitem

### PR DESCRIPTION
**Pull request by khalidmammadov**
Adds to PySpark API:
- sort by given columns
- getitem returns columns or DataFrame
- get attribute returns Column instead of duckdb.ColumnExpression

these changes are inline with PySpark API and was mainly copied from original implementation from Apache Spark project and fitted for duckdb

### CODE REVIEW

**Comment by khalidmammadov** [comment]
@Tishj can you please review?

**Review by Tishj — CHANGES_REQUESTED** [review]
Thanks for the PR, I left some minor comments

File: tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py [hunk 1]
```diff
@@ -263,14 +263,39 @@ def sort(
...
+
+        ascending = kwargs.get("ascending", True)
+
+        if isinstance(ascending, (bool, int)):
+            if not ascending:
+                columns = [c.desc() for c in columns]
+        elif isinstance(ascending, list):
+            columns = [c if asc else c.desc() for asc, c in zip(ascending, columns)]
+        else:
+            raise TypeError("Argument `ascending` should be a bool or"
```
[the remark this answers is not in the document]
Comment (khalidmammadov) [reply tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py:295]: I thought API should be consistent with PySpark hence the copy. Will mention next time.
Below replaced above error class. Not sure if it's needed to implement those classes or not at this point. Perhaps bit later when most of the main functionality is done.
```
        else:
            raise TypeError("Argument `ascending` should be a bool or"
                            f" list, got {type(ascending).__name__}.")
```
The error text is take from Spark's errors metadata.

[the remark this answers is not in the document]
Comment (khalidmammadov) [reply tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py]: Sorry, my bad. I didn't realise these classes have already been implemented. 
I have added them everywhere for sort method + test cases.

**Review by Tishj — APPROVED** [review]
Thanks for the contribution, greatly appreciated 👍

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**